helmclient.go

```diff
@@ -693,12 +720,68 @@ func getTillerImage(pod *corev1.Pod) (string, error) {
 
 	tillerImage := pod.Spec.Containers[0].Image
 	if tillerImage == "" {
-		return "", microerror.Mask(tillerImageNotFoundError)
+		return "", microerror.Mask(tillerImageInvalidError)
 	}
 
 	return tillerImage, nil
 }
 
+func isTillerOutdated(pod *corev1.Pod) error {
```

Comment:
What about extending return type to `(bool, error)`? currently we are trying to use `error` as response and error check as well.

Reply:
@tomahawk28 Generally we aim for fewer return params and we have matchers for our errors. So I'll use the error matcher to test for this error specifically.

https://github.com/giantswarm/helmclient/blob/f6ff1a2c6194749573ab2db9185bcc968614d6eb/error.go#L212

Our error handling is part of `microkit` and we have docs in `fmt`. Good to read if you haven't seen them already. https://github.com/giantswarm/fmt/blob/master/go/errors.md